src/init.cpp

```diff
@@ -1425,6 +1425,9 @@ bool AppInitMain()
                 pcoinsTip.reset();
                 pcoinsdbview.reset();
                 pcoinscatcher.reset();
+                if (fReset) {
```

Comment:
I don't think the conditional is necessary. Also, perhaps add a comment that the reset is necessary to avoid temporarily having two `CBlockTreeDB` objects?

Reply:
I put the conditional there because under normal circumstances `pblocktree.reset()` is useless. However I don't know what it does under the hood; maybe the performance overhead is negligible.